weasyl/profile.py

```diff
@@ -290,14 +290,16 @@ def select_relation(userid, otherid):
             (SELECT EXISTS (SELECT 0 FROM watchuser WHERE (userid, otherid) = (%i, %i))),
             (SELECT EXISTS (SELECT 0 FROM frienduser WHERE userid IN (%i, %i) AND otherid IN (%i, %i) AND settings !~ 'p')),
             (SELECT EXISTS (SELECT 0 FROM ignoreuser WHERE (userid, otherid) = (%i, %i))),
-            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE (userid, otherid) = (%i, %i) AND settings ~ 'p'))
-    """, [userid, otherid, userid, otherid, userid, otherid, userid, otherid, userid, otherid], ["single"])
+            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE (userid, otherid) = (%i, %i) AND settings ~ 'p')),
+            (SELECT EXISTS (SELECT 0 FROM watchuser WHERE (otherid, userid) = (%i, %i)))
+    """, [userid, otherid, userid, otherid, userid, otherid, userid, otherid, userid, otherid, otherid, userid], ["single"])
```

Comment:
I think I've managed to do something like that. It still uses d.execute (unfamiliar with engine), but query is formatted first, and it looks much nicer.